src/raw_mutex.rs

```diff
@@ -28,11 +28,30 @@ pub(crate) const TOKEN_NORMAL: UnparkToken = UnparkToken(0);
 // thread directly without unlocking it.
 pub(crate) const TOKEN_HANDOFF: UnparkToken = UnparkToken(1);
 
-const LOCKED_BIT: U8 = 1;
-const PARKED_BIT: U8 = 2;
+/// This bit is set in the `state` of a `RawMutex` when that mutex is locked by some thread.
+const LOCKED_BIT: U8 = 0b01;
+/// This bit is set in the `state` of a `RawMutex` when at least one thread is parked, waiting
+/// for the mutex to be unlocked.
+const PARKED_BIT: U8 = 0b10;
 
 /// Raw mutex type backed by the parking lot.
 pub struct RawMutex {
+    /// This atomic integer holds the current state of the mutex instance. Only the two lowest bits
+    /// are used. See `LOCKED_BIT` and `PARKED_BIT` for the bitmask for these bits.
+    ///
+    /// # State table:
+    ///
+    /// PARKED_BIT | LOCKED_BIT | Description
+    ///     0      |     0      | The mutex is not locked, nor is anyone waiting for it.
+    /// -----------+------------+------------------------------------------------------------------
+    ///     0      |     1      | The mutex is locked by exactly one thread. No other thread is
+    ///            |            | waiting for it.
+    /// -----------+------------+------------------------------------------------------------------
+    ///     1      |     0      | The mutex is not locked. There are parked threads waiting for it.
+    ///            |            | At least one of the parked threads are just about to be unparked.
+    /// -----------+------------+------------------------------------------------------------------
+    ///     1      |     1      | The mutex is locked by exactly one thread. At least one thread is
+    ///            |            | parked, waiting for the lock to become available.
```

Comment:
I would add an additional comment here because this is actually quite subtle:
- `PARKED_BIT` is only ever cleared when a bucket lock is held (i.e. in a `parking_lot_core` callback). This ensures that we never end up in a situation where there are parked threads but `PARKED_BIT` is not set (which would result in those threads potentially never getting woken up).
- We set `PARKED_BIT` before parking, so technically `PARKED_BIT` means "one or more thread is parked or about to park`. The `validate` callback is very important here since we need to ensure that `PARKED_BIT` is still set after we acquire the bucket lock (since a concurrent unpark may have cleared it).

Reply:
Great additions and clarifications. Copied some of it verbatim.